### PR TITLE
Replace XML solr requests with json solr request

### LINKED ISSUE #6465: Remove legacy XML parsing of solr output
Subtask of #6377

No longer needed with modern solr which output JSON, and quite a headache with nested documents

### PR DESCRIPTION
**Pull request by cdrini**
Closes #6465 . Refactor. Makes it easier to add editions, which will be a dynamic document. Loads more can be cleaned up here :/ But I think this is ok for now.

### Testing
- The `get_doc` function is only used in the search results UI ; confirm it looks the same between staging and testing
- Confirm .json results are working

### CODE REVIEW

File: openlibrary/plugins/worksearch/code.py [hunk 1]
```diff
@@ -217,48 +216,43 @@ def process_individual_sort(sort):
     return ','.join(process_individual_sort(s.strip()) for s in raw_sort.split(','))
 
 
-def read_author_facet(af):
-    # example input: "OL26783A Leo Tolstoy"
-    return re_author_facet.match(af).groups()
+def read_author_facet(author_facet: str) -> tuple[str, str]:
+    key, name = author_facet.split(' ', 1)
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:220]: Optional: Remove leading and trailing whitespace...
```suggestion
    key, name = author_facet.strip().split(' ', 1)
```

**Review by cclauss — APPROVED** [review]
LGTM.

File: openlibrary/plugins/worksearch/code.py (hunk 1 above)
Comment (cdrini) [reply openlibrary/plugins/worksearch/code.py:224]: Add the doctest :+1: There shouldn't be a need to strip here; the value coming from solr should not have any leading whitespace :+1:

**Comment by cdrini** [comment]
Thanks @cclauss :+1:

File: openlibrary/plugins/worksearch/code.py [hunk 2]
```diff
@@ -1272,7 +1224,7 @@ def work_search(
             facet=facet,
             spellcheck_count=spellcheck_count,
         )
-        response = json.loads(reply)['response'] or ''
+        response = reply['response'] or ''
```
Comment (jimchamp) [inline openlibrary/plugins/worksearch/code.py:1227]: `None` case is not handled here.  This is causing my local instance's homepage to error on first load, resulting in no content (aside from the page header and footer).

Comment (jimchamp) [reply openlibrary/plugins/worksearch/code.py:1227]: This sort of worked:
`response = reply.get('response', '') if reply else ''`

The issue with this is that `response` is being subscripted like a `dict` later in this function, leading to more errors.

@cdrini, what are sensible defaults for `response.get('numFound')` and `response.get('docs')`?  Using `0` and `{}` respectively causes the homepage to render locally without the classics carousel:

Screenshot from 2022-05-31 19-27-20

Comment (cclauss) [reply openlibrary/plugins/worksearch/code.py:1227]: #6616

**Pull request merged**